resources/images/filters.go

```diff
@@ -33,6 +35,47 @@ func (*Filters) Overlay(src ImageSource, x, y interface{}) gift.Filter {
 	}
 }
 
+// Text creates a filter that draws text with the given options.
+func (*Filters) Text(text string, options ...interface{}) gift.Filter {
+	tf := textFilter{
+		text:  text,
+		color: "#000000",
+		size:  20,
+		x:     1,
+		y:     1,
+	}
+
+	if len(options) > 0 {
+		opt, err := maps.ToStringMapE(options[0])
+		if err != nil {
+			panic(err)
+		}
+
+		for option, v := range opt {
+			switch option {
+			case "color":
+				tf.color = cast.ToString(v)
+			case "size":
+				tf.size = cast.ToFloat64(v)
+			case "x":
+				tf.x = cast.ToInt(v)
+			case "y":
+				tf.y = cast.ToInt(v)
+			}
+
+		}
+
+		if w, ok := opt["font"].(fontSource); ok {
```

Comment:
OK, I admit, it would be useful to set a custom font, but then this needs some more work. As it is now, I'm pretty sure that `HashString` will return a new value on every build/restart, and that doesn't work.

I'm doing some related work in this area, so I suggest that we for this case make it simple:

* Fetch the font if set
* Replace the `font` in options with `font.Key` (there is an interface)
* Create the options hash

Reply:
I have removed the font for now, but custom fonts make the Text filter much more useful; this would be a requirement for many users.

Not sure if I understand you suggestion above.

Reply:
I suggest that we complete this PR without the `font` attribute -- I can create suggested PR for that after and see how that works. I'm planning a Hugo release in the next few days, and this would be a great feature to get better OpenGraph images on the Hugo docs site.

Reply:
Removed font attribute as suggested